### PR TITLE
Updated tutorial docs

### LINKED ISSUE #1208: Enhance tutorial by adding Prerequisites, Understanding the example code, What is an N-dimensional array?
Going through the [Zarr tutorial](https://zarr.readthedocs.io/en/stable/tutorial.html), I was thinking it will be nice to have a 
- Prerequisites section: Which tells the user the basic skills necessary for using Zarr, the dependencies needed to use.
- Understanding the example code: Which contains guidelines on how to read the code snippets in the examples
- What is an N-dimensional array: Which contains definitions and samples of N-dimensional array

### PR DESCRIPTION
**Pull request by DON-BRAN**
fixes #1208

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Review by joshmoore — COMMENTED** [review]
Looks good structurally, though I'm not the best person to say if these URLs are the best starting points. Copying a screenshot from https://zarr--1217.org.readthedocs.build/en/1217/ in case anyone wants to give it a quick glance.

Screen Shot 2022-10-25 at 16 31 57

File: docs/tutorial.rst [hunk 1]
```diff
@@ -3,11 +3,50 @@
...
 ========
 
+.. _tutorial_what_is_zarr?:
+
+What is Zarr?
+-------------
+
+Zarr is a format used to create N-dimensional arrays with any NumPy `dtype <https://numpy.org/doc/stable/reference/arrays.dtypes.html#arrays-dtypes>`_
+
+- Chunk arrays along any dimension
```
[the remark this answers is not in the document]
Comment (DON-BRAN) [reply docs/tutorial.rst:13]: Thank you @joshmoore for the feedback. I will make updates 🙂

[the remark this answers is not in the document]
Comment (DON-BRAN) [reply docs/tutorial.rst]: Hi @joshmoore, I noticed anytime i build locally I see the bullet points well represented

image

But when it builds on the server the bullet points get extra white spaces between them

image

Found this on stack overflow https://stackoverflow.com/questions/67542699/readthedocs-sphinx-not-rendering-bullet-list-from-rst-file

[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/tutorial.rst]: "Edit: that CSS bug is in sphinx_rtd_theme https://github.com/readthedocs/sphinx_rtd_theme/issues/1115"

Interesting. One option then would be to move to another them, e.g. https://github.com/pydata/pydata-sphinx-theme

cc: @MSanKeys963

[the remark this answers is not in the document]
Comment (DON-BRAN) [reply docs/tutorial.rst]: > "Edit: that CSS bug is in sphinx_rtd_theme [readthedocs/sphinx_rtd_theme#1115](https://github.com/readthedocs/sphinx_rtd_theme/issues/1115)"
> 
> Interesting. One option then would be to move to another them, e.g. https://github.com/pydata/pydata-sphinx-theme
> 
> cc: @MSanKeys963

I will like to work on the theme migration if that's voted :)

[the remark this answers is not in the document]
Comment (joshmoore) [reply docs/tutorial.rst]: I'd suggest to go ahead and try it out. We can always revert.

**Comment by jhamman** [comment]
This has unfortunately gone stale. Will it be revived or should we close this?

**Comment by jhamman** [comment]
Closing as stale. We'll keep the original feature request open.

**Pull request closed without merging**